src/client/src/tscSQLParser.c

```diff
@@ -3183,7 +3184,7 @@ int32_t addExprAndResultField(SSqlCmd* pCmd, SQueryInfo* pQueryInfo, int32_t col
             for (int32_t j = 0; j < tscGetNumOfColumns(pTableMetaInfo->pTableMeta); ++j) {
               index.columnIndex = j;
               SStrToken t = {.z = pSchema[j].name, .n = (uint32_t)strnlen(pSchema[j].name, TSDB_COL_NAME_LEN)};
-              setResultColName(name, pItem, cvtFunc.originFuncId, &t, true);
+              setResultColName(name, finalResult, pItem, cvtFunc.originFuncId, &t, true);
```

Comment:
Keep the column name for results of TSDB_FUNC_{FIRST,LAST,LAST_ROW,SPREAD,INTERP} only when they appear as the root of EXPR tree, which is indicated by finalResult as true. For example, select first(f1) from t; instead of select first(f1) + 1 from t;

Reply:
For example,
```text
taos> select first(int_col) from t1;
   int_col   |
==============
  1947932226 |
Query OK, 1 row(s) in set (0.002255s)

taos> select first(int_col) + 1 from t1;
    first(int_col) + 1     |
============================
      1947932227.000000000 |
Query OK, 1 row(s) in set (0.002326s)

taos> 
```